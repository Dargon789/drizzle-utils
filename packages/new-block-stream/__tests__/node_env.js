/**
 * @jest-environment node
 */
const Ganache = require("ganache-core");
const Web3 = require("web3");
// const { take, finalize, tap, toArray } = require("rxjs/operators");

const compile = require("./utils/compile");
const createNewBlock$ = require("../index");

jest.setTimeout(20000);

describe("new-block-stream tests in node environment", () => {
  let provider;
  let web3;
  let accounts;
  let contractInstance;
  //   let artifact;

  beforeAll(async () => {
    // 1. Compile contract artifact
    const { SimpleStorage } = await compile("SimpleStorage.sol");

    // 2. Spawn Ganache test blockchain
    provider = Ganache.provider({ seed: "drizzle-utils" });
    web3 = new Web3(provider);
    accounts = await web3.eth.getAccounts();

    // 3. Create initial contract instance
    const instance = new web3.eth.Contract(SimpleStorage.abi);

    // 4. Deploy contract and get new deployed Instance
    const deployedInstance = await instance
      .deploy({ data: SimpleStorage.bytecode })
      .send({ from: accounts[0], gas: 150000 });

    // Note: deployed address located at `deployedInstance._address`
    contractInstance = deployedInstance;

    // artifact = {
    //   ...SimpleStorage,
    //   // truffle-decoder needs this in artifact
    //   networks: {
    //     "4447": {
    //       address: deployedInstance._address,
    //     },
    //   },
    // };
  });

  afterAll(async () => {
    provider.close();
  });

  test("ensure test suite setup works", async () => {
    // Get old value, set a new value, check to see if it was set correctly
    const oldVal = await contractInstance.methods.get().call();
    await contractInstance.methods.set(5).send({ from: accounts[0] });
    const newVal = await contractInstance.methods.get().call();

    expect(oldVal).toBe("0");
    expect(newVal).toBe("5");
  });

  test("createNewBlock$ function exists", () => {
    expect(createNewBlock$).toBeDefined();
  });
});
