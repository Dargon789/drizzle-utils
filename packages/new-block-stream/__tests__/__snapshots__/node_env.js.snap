// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new-block-stream tests in node environment fromPolling can track blocks 1`] = `
Object {
  "difficulty": "0",
  "extraData": "0x",
  "gasLimit": 6721975,
  "gasUsed": 41711,
  "hash": Anything,
  "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "miner": "0x0000000000000000000000000000000000000000",
  "mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "nonce": "0x0000000000000000",
  "number": 2,
  "parentHash": Anything,
  "receiptsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
  "sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
  "size": 1000,
  "stateRoot": "0x9182e4bbb0559e4821e881204cca9ba38ee1eb65999b6d34c6ebba00ff6cc376",
  "timestamp": Anything,
  "totalDifficulty": "0",
  "transactions": Array [
    Object {
      "blockHash": Anything,
    },
  ],
  "transactionsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
  "uncles": Array [],
}
`;

exports[`new-block-stream tests in node environment fromPolling can track blocks 2`] = `
Object {
  "difficulty": "0",
  "extraData": "0x",
  "gasLimit": 6721975,
  "gasUsed": 41711,
  "hash": Anything,
  "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  "miner": "0x0000000000000000000000000000000000000000",
  "mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "nonce": "0x0000000000000000",
  "number": 4,
  "parentHash": Anything,
  "receiptsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
  "sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
  "size": 1000,
  "stateRoot": "0x26c20d98f961ecb37857143c22ac3e90c9e214a61a8511763ca44d4e110f6d3d",
  "timestamp": Anything,
  "totalDifficulty": "0",
  "transactions": Array [
    Object {
      "blockHash": Anything,
    },
  ],
  "transactionsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
  "uncles": Array [],
}
`;
